code/sha3/Hacl.Impl.SHA3.fst

```diff
@@ -377,6 +377,28 @@ let storeState rateInBytes s res =
       copy res (sub block 0ul rateInBytes)
     )
 
+(* 
+   Lemma_equivalence_state_rate proves that it is equivalent
+   to allocate a bigger temporary buffer and then use a part of it 
+   compared to allocation of the exact size buffer
+*)
+
```

Comment:
This lemma is overly specific. Can't it be omitted or inlined in the definition of `absorb_next` and `absorb_last`? If not, can you at least generalize it by dropping the refinement in the parameter and generalizing the length and type?

Reply:
I will try! Thanks for the suggestion.